src_template/control.h

```diff
@@ -35,10 +61,63 @@ typedef struct {
 // ----------------------------------------------------------------------------
 // functions control.c needs to implement (will be called from main.c)
 
+/**
+ * @brief Called implicitly by Multipass at startup if the hardware flash is
+ *        new and there are no presets stored to flash. This function is the
+ *        applications opportunity to initialize the flash with application
+ *        appropriate preset structs and shared data in default state. Once
+ *        initialized, Multipass will no longer call this function at startup.
+ * 
+ * @see is_flash_new()
+ * @see store_preset_to_flash()
+ * @see store_shared_data_to_flash()
+ * @see store_preset_index()
+ * @see preset_data_t
+ * @see preset_meta_t
+ * @see shared_data_t
+ */
 void init_presets(void);
+
+/**
+ * @brief Called implicitly by Multipass at application startup. This function
+ *        is the applications opportunity to load shared data, load a preset and
+ *        meta data, set up any initial application values, and set up timers.
+ * 
+ * @see load_shared_data_from_flash()
+ * @see load_preset_from_flash()
+ * @see load_preset_meta_from_flash()
+ */
 void init_control(void);
+
+/**
+ * @brief Implement event handling code for your application here. This function
+ *        is called by Multipass when an event happens. Here your application
+ *        has the opportunity to respond to these events. See `interface.h`
+ *        events macro definition section for a list of event identifier macros,
+ *        data, and length documentation.
+ * 
+ * @param event Event identifier
+ * @param data Pointer to event data
+ * @param length Length of event data
+ */
 void process_event(u8 event, u8 *data, u8 length);
+
+/**
+ * @brief Implement Monome grid LED rendering code for your application here. If
+ *        your application does not use the Monome grid, you can leave this
+ *        function blank. This function is called by Multipass at a rate of
+ *        MONOME_REFRESH_INTERVAL if a Monome grid is connected to the hardware.
```

Comment:
the function will only be called after `refresh_grid()` was executed, and it will check that at a rate of `MONOME_REFRESH_INTERVAL` (same for `render_arc` below)